### PR TITLE
Fix error in core fact script, support global 'unsafe_writes' parameter

### PR DESCRIPTION
**Pull request by drybjed**

### CODE REVIEW

**Comment by drybjed** [comment]
Fixes #27.

**Comment by drybjed** [comment]
Requesting review from @ypid

File: templates/etc/ansible/facts.d/core.fact.j2 [hunk 1]
```diff
@@ -77,7 +78,8 @@ if output['admin_users']:
     if isinstance(output['admin_users'], basestring):
         admin_users.add(output['admin_users'])
     elif isinstance(output['admin_users'], list):
-        admin_users.extend(output['admin_users'])
+        for element in output['admin_users']:
+            admin_users.add(element)
```
Comment (ypid) [inline templates/etc/ansible/facts.d/core.fact.j2]: I would propose:

```Python
admin_users.union(output['admin_users'])
```
instead same as @cultcom already did in https://github.com/debops/ansible-core/issues/27. Looks cleaner.

Comment (drybjed) [inline templates/etc/ansible/facts.d/core.fact.j2]: Right, I haven't seen the edit. I'll change it then, thanks.

Comment (drybjed) [inline templates/etc/ansible/facts.d/core.fact.j2]: @ypid I changed the script to use `union()` but the list is not merged back into the set. Can you check if that really works?

Comment (ypid) [inline templates/etc/ansible/facts.d/core.fact.j2]: You are right. Seems I checked not careful enough. `set().union()` returns a new set.
```Python
`admin_users.update(set(output['admin_users']))
```
does the desired.

Comment (cultcom) [inline templates/etc/ansible/facts.d/core.fact.j2]: you are right, "update" looks better  here.

**Comment by cultcom** [comment]
looks good, but I think you do not need to cast the extra "set()" arround it. "update" accepts lists as well:

```

In [1]: a = set()

In [2]: b = ["a", "b", "b", "c"]

In [3]: a.update(b)

In [4]: a
Out[4]: {'a', 'b', 'c'}

In [5]: 
```

**Comment by ypid** [comment]
@cultcom I was aware that this does work but I wanted to be explicit. But the shorter variant is also widely used and I will probably prefer this from now on. Thanks.

Ref: https://stackoverflow.com/questions/28845284/add-vs-update-in-set-operations-in-python
Ref: https://docs.python.org/3/library/stdtypes.html#set.update

**Pull request merged**